ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/management-api-gnmi.j2

```diff
@@ -36,6 +36,38 @@ management api gnmi
 {%                 endif %}
 {%             endfor %}
 {%         endif %}
+{%         if management_api_gnmi.transport.grpc_tunnel is arista.avd.defined %}
+{%             for transport in management_api_gnmi.transport.grpc_tunnel %}
+{%                 if transport.name is arista.avd.defined %}
+   transport grpc-tunnel {{ transport.name }}
+{%                 endif %}
+{%                 if transport.shutdown is arista.avd.defined(true) %}
+      shutdown
+{%                 elif transport.shutdown is arista.avd.defined(false) %}
+      no shutdown
+{%                 endif %}
+{%                 if transport.vrf is arista.avd.defined %}
+      vrf {{ transport.vrf }}
+{%                 endif %}
+{%                 if transport.tunnel_ssl_profile is arista.avd.defined %}
+      tunnel ssl profile {{ transport.tunnel_ssl_profile }}
+{%                 endif %}
+{%                 if transport.gnmi_ssl_profile is arista.avd.defined %}
+      gnmi ssl profile {{ transport.gnmi_ssl_profile }}
+{%                 endif %}
+{%                 if transport.destination is arista.avd.defined %}
+      destination {{ transport.destination.address }} port {{ transport.destination.port }}
```

Comment:
This template could fail if address or port are missing:

```
management_api_gnmi:
  transport:
    grpc_tunnel:
      - name: TEST
         destination:
           address: 42.42.42.42
```

I have updated the schema in comments so that if `destination` is defined then both address and port are required which means that this will never happen